sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/TokenValidatorEZAuth.cs

```diff
@@ -1,37 +1,35 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents.Framework;
-using System;
 using System.Collections.Generic;
-using System.Linq;
 using System.Net.Http;
 using System.Threading.Tasks;
 
+using Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents.Framework;
+
 namespace Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents
 {
     internal class TokenValidatorEZAuth : TokenValidator
     {
-        internal override Task<(bool Valid, Dictionary<string, string> Claims)> GetClaimsAndValidate(HttpRequestMessage request, ConfigurationManager configurationManager)
+        internal override Task<Dictionary<string, string>> ValidateAndGetClaims(
+            HttpRequestMessage request,
+            ConfigurationManager configurationManager)
         {
             Dictionary<string, string> Claims = new Dictionary<string, string>();
-            try
+            string principal = request.Headers.DecodeBase64(ConfigurationManager.HEADER_EZAUTH_PRINCIPAL);
+            AuthenticationEventJsonElement jPrincipal = new AuthenticationEventJsonElement(principal);
+
+            foreach (AuthenticationEventJsonElement jVal in jPrincipal.GetPropertyValue<AuthenticationEventJsonElement>("claims").Elements)
             {
-                string principal = request.Headers.DecodeBase64(ConfigurationManager.HEADER_EZAUTH_PRINCIPAL);
-                AuthenticationEventJsonElement jPrincipal = new AuthenticationEventJsonElement(principal);
+                Claims.Add(jVal.GetPropertyValue("typ"), jVal.GetPropertyValue("val"));
```

Comment:
Should "typ", "val" and "ver" be defined at the top?

Reply:
nah, this is a very specific to Claims check property. We will most like update this logic next time.